### PR TITLE
Implement UI for block index

### LINKED ISSUE #746: Implement UI for block index

### PR DESCRIPTION
**Pull request by Benzbeeb**
Fixes #746

### CODE REVIEW

File: scan/src/pages/BlockIndexPage.re [hunk 1]
```diff
@@ -43,92 +43,102 @@ module Styles = {
...
+let make = (~height: int) =>
+  {
+    let (limit, setLimit) = React.useState(_ => 10);
+    let txsOpt = TxHook.atHeight(height, ~limit, ());
+    let infoOpt = React.useContext(GlobalContext.context);
+    let blockOpt = BlockHook.atHeight(height);
+    let%Opt info = infoOpt;
+    let%Opt block = blockOpt;
+    let validators = info.validators;
+    let moniker = BlockHook.Block.getProposerMoniker(block, validators);
```
Comment (evilpeach) [inline scan/src/pages/BlockIndexPage.re]: Where the moniker are used?

File: scan/src/components/blockIndexTxsTable.re [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  let container = style([width(`px(68))]);
+  let hashContainer = style([maxWidth(`px(140))]);
+  let paddingTopContainer = style([paddingTop(`px(5))]);
+  let statusContainer =
+    style([maxWidth(`px(95)), display(`flex), flexDirection(`row), alignItems(`center)]);
+  let logo = style([width(`px(20)), marginLeft(`auto), marginRight(`px(15))]);
+};
+[@react.component]
+let make = (~txs: list(TxHook.Tx.t)) => {
+  // let x = txs.
```
Comment (taobun) [inline scan/src/components/blockIndexTxsTable.re]: Remove this line

File: scan/src/pages/BlockIndexPage.re (hunk 1 above)
Comment (evilpeach) [inline scan/src/pages/BlockIndexPage.re]: Let's code like this
```
  let monikerOpt = {
    let%Opt info = infoOpt;
    let%Opt block = blockOpt;
    let validators = info.validators;
    Some(BlockHook.Block.getProposerMoniker(block, validators));
  };
```

File: scan/src/pages/BlockIndexPage.re [hunk 3]
```diff
@@ -43,92 +43,102 @@ module Styles = {
...
-         code=true
-         nowrap=true
-         ellipsis=true
-       />
-     | None => <Text value="in the future" size=Text.Xxl />
-     }}
-    <VSpacing size=Spacing.lg />
-    <Row>
-      <Col size=1.8>
+    Some(
```
Comment (evilpeach) [inline scan/src/pages/BlockIndexPage.re]: remove `Some`

File: scan/src/pages/BlockIndexPage.re [hunk 4]
```diff
@@ -43,92 +43,102 @@ module Styles = {
...
+                  ? React.null : <LoadMore onClick={_ => setLimit(oldLimit => oldLimit + 10)} />}
+               <VSpacing size=Spacing.xl />
+             </>
+           }
+         | _ => <VSpacing size={`px(280)} />
+         }}
+      </div>,
+    );
+  }
+  ->Belt_Option.getWithDefault(React.null);
```
Comment (evilpeach) [inline scan/src/pages/BlockIndexPage.re]: remove this one

File: scan/src/pages/BlockIndexPage.re [hunk 5]
```diff
@@ -43,92 +43,102 @@ module Styles = {
...
-           {txs->Belt_List.size < limit
-              ? React.null : <LoadMore onClick={_ => setLimit(oldLimit => oldLimit + 10)} />}
-           <VSpacing size=Spacing.xl />
-         </>
-       }
-     | _ => <VSpacing size={`px(280)} />
-     }}
-  </div>;
-};
+               | None => <InfoHL info={InfoHL.Text("?")} header="PROPOSED BY" />
```
Comment (evilpeach) [inline scan/src/pages/BlockIndexPage.re]:
```
switch (monikerOpt) {
   | Some(moniker) => <InfoHL info={InfoHL.Text(moniker)} header="PROPOSED BY" />
   | None => <InfoHL info={InfoHL.Text("?")} header="PROPOSED BY" />
}``

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: scan/src/pages/BlockIndexPage.re (hunk 3 above)
Comment (evilpeach) [reply scan/src/pages/BlockIndexPage.re]: If you use `Some` and `let%Opt`, the switch inside render won't work.

**Review by taobun — CHANGES_REQUESTED** [review]

File: scan/src/pages/BlockIndexPage.re [hunk 6]
```diff
@@ -43,62 +45,80 @@ module Styles = {
...
             nowrap=true
             color=Colors.mediumGray
             block=true
+            spacing={Text.Em(0.06)}
           />
           <div className=Styles.seperatedLine />
           {switch (blockOpt) {
-           | Some(block) => <TimeAgos time={block.timestamp} size=Text.Lg weight=Text.Regular />
+           | Some(block) =>
+             <Text
```
Comment (evilpeach) [inline scan/src/pages/BlockIndexPage.re]: use `<TypeID.Block id={ID.Block.ID(height)} />` instead.

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — DISMISSED** [review]

File: scan/src/components/ReportTable.re [hunk 7]
```diff
@@ -48,8 +48,8 @@ let make = (~reports: list(RequestHook.Report.t)) => {
            ->Belt_List.get(0)
            ->Belt_Option.getWithDefault("Unknown");
          <TBody key={txHash |> Hash.toHex}>
-           <Row alignItems=Css.flexStart>
 
+             <Row alignItems=Css.flexStart>
```
Comment (taobun) [inline scan/src/components/ReportTable.re]: No diff in this file?

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**